sql/src/main/java/io/crate/analyze/AlterTableRerouteAnalyzer.java

```diff
@@ -89,5 +120,30 @@ public RerouteAnalyzedStatement visitRerouteCancelShard(RerouteCancelShard node,
             return new RerouteCancelShardAnalyzedStatement(
                 context.tableInfo, context.partitionProperties, node.shardId(), node.nodeId(), node.properties());
         }
+
+        @Override
+        public RerouteAnalyzedStatement visitReroutePromoteReplica(PromoteReplica promoteReplica, Context context) {
+            HashMap<String, Expression> properties = new HashMap<>(promoteReplica.properties().properties());
+            Expression acceptDataLossExpr = properties.remove(PromoteReplica.Properties.ACCEPT_DATA_LOSS);
+            if (!properties.isEmpty()) {
+                throw new IllegalArgumentException("Unsupported options provided to REROUTE PROMOTE REPLICA: " + properties.keySet());
+            }
+            var exprCtx = new ExpressionAnalysisContext();
+            var expressionAnalyzer = new ExpressionAnalyzer(
+                functions, context.txnCtx, context.paramTypeHints, FieldProvider.UNSUPPORTED, null);
+
+            Symbol acceptDataLoss = acceptDataLossExpr == null
+                ? Literal.BOOLEAN_FALSE
+                : expressionAnalyzer.convert(acceptDataLossExpr, exprCtx);
+            Symbol shardId = expressionAnalyzer.convert(promoteReplica.shardId(), exprCtx);
+            Symbol nodeId = expressionAnalyzer.convert(promoteReplica.nodeId(), exprCtx);
+            return new PromoteReplicaStatement(
+                context.tableInfo,
+                context.partitionProperties,
```

Comment:
Note that this is a bit of a inconsistency as `partitionProperties` are unresolved `Expression`s. I'll follow up on this to align it.

I also derived from the existing pattern of the Reroute command processing, they didn't do symbol conversions at all.
If we ever want to get rid of the bound analyzer and support the parameter description message in the postgres protocol for all statements, I think we should/have to do expression -> symbol conversions.  That's why I opted for this pattern here now.